src/policy/satisfy.rs

```diff
@@ -91,139 +95,203 @@ impl<Pk: ToXOnlyPubkey> Satisfier<Pk> for elements::LockTime {
     }
 }
 
+#[derive(Debug)]
+pub enum SatisfierError {
+    /// A satisfying witness for the policy could not be produced.
+    Unsatisfiable,
+    /// An assembly program was missing.
+    ///
+    /// The satisfier needs to know all assembly programs, **even those that are not executed**!
+    MissingAssembly(Cmr),
+    /// An assembly program failed to run on the Bit Machine.
+    ///
+    /// This can happen when the [`Satisfier::lookup_asm_program`] method is incorrectly implemented.
+    AssemblyFailed(crate::bit_machine::ExecutionError),
+}
+
+/// A witness expression wrapped in `Ok` if satisfaction was successful
+/// or wrapped in `Err` if satisfaction failed.
+///
+/// This information is necessary for the policy satisfier to select the correct
+/// disjunction (`or`) and threshold (`thres`) branches.
```

Comment:
In c5b5f0299adf43012ee054899850ce31c1121003:

nit: `threshold` should be shortend to `thresh`.